### ISSUE TITLE
DataFusion weekly project plan (Andrew Lamb) - May 20, 2024

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#10579]
Follow on to https://github.com/apache/datafusion/issues/10482

My (personal) North ⭐ : 1000 projects are built using DataFusion 📈 

**It would be great for other contributors to DataFusion who plan non trivial work could try to make them visible somehow as well** 🙏  (feel free to copy / modify the format)

## My Highlights from Last Week
* Completed the Expr --> SQL (https://github.com/apache/datafusion/issues/9726)  thanks to @xinlifoobar @reswqa, @goldmedal @edmondop  and others
* @berkaysynnada and @ozankabak made the UDF analysis framework much more general: https://github.com/apache/datafusion/pull/10504
* @jayzhan211 is cranking along extracting Aggregates from the core https://github.com/apache/datafusion/pull/10493
* (a week late) @comphead and @mustafasrepo joined the PMC, @jonahgao joined as a committer. See the mailing list threads [here](https://lists.apache.org/thread/hr8yf2pb573lm2sm5cgvtr126n1bm2v3) [here](https://lists.apache.org/thread/j07fbpmtjxcx13fskjo71vwgxxv6snyt) and [here](https://lists.apache.org/thread/xn2tgnzb59qwmwxqqdjgjjnplh358gk3).
* @comphead and @viirya are improving SortMergeJoin https://github.com/apache/datafusion/pull/10495
* @NGA-TRAN  and I made significant progress making an API to work with parquet statistics https://github.com/apache/datafusion/issues/10453

## My (personal) plans for this week
TLDR is I plan to keep trying to make parquet support better (documentation and examples)
- [x] Document `ParquetExec` more (so it is clear / easier to understand what is already supported)
- [x] Help @NGA-TRAN  with https://github.com/apache/datafusion/issues/10453
- [ ] Complete https://github.com/apache/datafusion/issues/10546
- [ ] https://github.com/apache/datafusion/issues/10480
- [ ] https://github.com/apache/datafusion/issues/10403
- [ ] Help  @tustvold  complete https://github.com/apache/arrow-rs/issues/5736 (likely by reviewing)

## Project Queue (list of future projects)
- [ ] https://github.com/apache/arrow-datafusion/issues/9929 (likely not this week)
- [ ] (Stretch) https://github.com/apache/datafusion/issues/9873 copies (last part of  https://github.com/apache/arrow-datafusion/issues/9637)
- [ ] In general, help https://github.com/apache/arrow-rs/issues/5374 more actively as it seems to be stuck

## Projects I plan to help actively help review / plan
- [ ] https://github.com/apache/arrow-datafusion/issues/9494

# Algorithm for (my) prioritizing PR reviews
Note there are many [committers](https://projects.apache.org/committee.html?datafusion) who can and do review and merge PRs, so this is not the priorities of the project as a whole, just the approximate algorithm I use to prioritize my own time.

Priority:
1. Bug fixes (where something is just incorrect), especially regressions (where it used to work and now does not)
2. Improvements directly related to features needed for InfluxDB (my employer)
3. Documentation and test improvements (I view these as very strategically important)
4. PRs that I think are strategically important
5. Other new features / additions to functionality (note this is the lowest)

The top strategically important projects in my head are:

* Anything that makes it easier to use DataFusion as a user (docs, examples, better APIs, etc), including [Getting started guide for new users (who want to use DataFusion in their project) ](https://github.com/apache/arrow-datafusion/issues/7014) 
* Anything that improves DataFusion's quality such as bug fixes, and improved / expanded tests, etc
* Making function packages modular and easier to mix/match 
* Improved performance (with benchmarks)

Thus, if you are interested in contributing to DataFusion and are interested in a fast turn around time I would recommend looking into bug fixes / test improvements / documentation or the projects named above. 

If you propose adding new functionality, especially if the PR is large/complex and not connected to a wider need, the review cycle will likely be longer. You can make it a shorter cycle by looking at the comments on other recent PRs and following the same model (e.g. ensure there are tests in sqllogictest for example, the CI passes, includes documentation, etc)

## Background
The idea of this ticket is make my plans for DataFusion visible, largely for my own personal organizational needs, but also to:
1. Communicate / coordinate in the community 
2. Help provide an interesting summary of what is happening in DataFusion this week

**Comment by alamb** [comment]
Next week: https://github.com/apache/datafusion/issues/10699
